python/cudf/cudf/core/column/categorical.py

```diff
@@ -1394,31 +1395,17 @@ def reorder_categories(
             )
         return self._set_categories(new_categories, ordered=ordered)
 
-    def as_ordered(self):
-        out_col = self
-        if not out_col.ordered:
-            out_col = column.build_categorical_column(
-                categories=self.categories,
-                codes=self.codes,
-                mask=self.base_mask,
-                size=self.base_size,
-                offset=self.offset,
-                ordered=True,
-            )
-        return out_col
-
-    def as_unordered(self):
-        out_col = self
-        if out_col.ordered:
-            out_col = column.build_categorical_column(
+    def as_ordered(self, ordered: bool):
+        if self.dtype.ordered != ordered:
```

Comment:
I'd rather flip the if/else and structure this as an early exit.

If no work is needed (`self.dtype.ordered == ordered`), return `self` immediately.

Otherwise return the longer result of `build_categorical_column`.